src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -1,23 +1,214 @@
+    }
+
+    /// <summary>
+    /// Tries to add the datasource to the DataSourceNameToDataSource dictionary.
+    /// </summary>
+    /// <param name="dataSourceName">dataSourceName.</param>
+    /// <param name="dataSource">dataSource.</param>
+    /// <returns>True indicating success, False indicating failure.</returns>
+    public bool TryAddDataSource(string dataSourceName, DataSource dataSource)
+    {
+        return _dataSourceNameToDataSource.TryAdd(dataSourceName, dataSource);
+    }
+
+    /// <summary>
+    /// Updates the DataSourceNameToDataSource dictionary with the new datasource.
+    /// </summary>
+    /// <param name="dataSourceName">Name of datasource</param>
+    /// <param name="dataSource">Updated datasource value.</param>
+    /// <exception cref="DataApiBuilderException">Not found exception if key is not found.</exception>
+    public void UpdateDataSourceNameToDataSource(string dataSourceName, DataSource dataSource)
+    {
+        CheckDataSourceNamePresent(dataSourceName);
+        _dataSourceNameToDataSource[dataSourceName] = dataSource;
+    }
+
+    /// <summary>
+    /// Removes the datasource from the DataSourceNameToDataSource dictionary.
+    /// </summary>
+    /// <param name="dataSourceName">DataSourceName.</param>
+    /// <returns>True indicating success, False indicating failure.</returns>
+    /// <exception cref="DataApiBuilderException">Not found exception if key is not found.</exception>
+    public bool RemoveDataSource(string dataSourceName)
+    {
+        CheckDataSourceNamePresent(dataSourceName);
+        return _dataSourceNameToDataSource.Remove(dataSourceName);
+    }
+
+    /// <summary>
+    /// Gets datasourceName from EntityNameToDatasourceName dictionary.
+    /// </summary>
+    /// <param name="entityName">entityName</param>
+    /// <returns>DataSourceName</returns>
+    public string GetDataSourceNameFromEntityName(string entityName)
+    {
+        CheckEntityNamePresent(entityName);
+        return _entityNameToDataSourceName[entityName];
+    }
+
+    /// <summary>
+    /// Gets datasource using entityName.
+    /// </summary>
+    /// <param name="entityName">entityName.</param>
+    /// <returns>DataSource using EntityName.</returns>
+    public DataSource GetDataSourceFromEntityName(string entityName)
+    {
+        CheckEntityNamePresent(entityName);
+        return _dataSourceNameToDataSource[_entityNameToDataSourceName[entityName]];
+    }
+
+    /// <summary>
+    /// Add entity to the EntityNameToDataSourceName dictionary.
+    /// </summary>
+    /// <param name="entityName">EntityName</param>
+    /// <param name="dataSourceName">DatasourceName.</param>
+    /// <returns>True indicating success, False indicating failure.</returns>
+    /// <exception cref="DataApiBuilderException">Not found exception if key is not found.</exception>
+    public bool TryAddEntity(string entityName, string dataSourceName)
+    {
+        CheckDataSourceNamePresent(dataSourceName);
```

Comment:
Functions starting with Try e.g. here TryAddEntity usually shouldnt throw exceptions. Either change the name to AddEntity or avoid throwing exception, just return false.